DigitalLearningSolutions.Web/Controllers/ChangePasswordController.cs

```diff
@@ -28,30 +28,24 @@ public IActionResult Index()
         [HttpPost]
         public async Task<IActionResult> Index(ChangePasswordViewModel model)
         {
-            if (!ModelState.IsValid)
-            {
-                return View(model);
-            }
-
-            var currentPassword = model.CurrentPassword!;
-
             var adminId = User.GetAdminId();
             var delegateId = User.GetCandidateId();
-
-            var verifiedLinkedUsersAccounts =
-                userService.GetVerifiedLinkedUsersAccounts(adminId, delegateId, currentPassword);
-
-            var passwordIsInvalid = !verifiedLinkedUsersAccounts.Any();
-
-            if (passwordIsInvalid)
+            if (model.CurrentPassword != null && !userService.IsPasswordValid(adminId, delegateId, model.CurrentPassword))
             {
                 ModelState.AddModelError(
-                    nameof(model.CurrentPassword),
+                    nameof(ChangePasswordViewModel.CurrentPassword),
                     CommonValidationErrorMessages.IncorrectPassword
                 );
+            }
+
+            if (!ModelState.IsValid)
+            {
                 return View(model);
             }
 
+            var verifiedLinkedUsersAccounts =

```

Comment:
Given that we need the verifiedLinkedUserAccounts anyway, could we do something like:
```
var verifiedLinkedUsersAccounts = string.IsNullOrEmpty(model.CurrentPassword)
                ? new UserAccountSet()
                : userService.GetVerifiedLinkedUsersAccounts(adminId, delegateId, model.CurrentPassword!);
if (verifiedLinkedUsersAccounts.Any()) {
// Add errors
}
```
Then we can continue without having to make 2x calls to getting linked accounts. Will probably be more work than worth to avoid double calls on the MyAccountController though.